src/components/Grid/Grid.tsx

```diff
@@ -0,0 +1,140 @@
+import React, {
+  Children as ReactChildren,
+  createElement,
+  cloneElement,
+  CSSProperties,
+  Fragment,
+  useEffect,
+  useMemo,
+  useState,
+  createRef,
+  PropsWithChildren,
+} from "react";
+import { forwardRef } from "react";
+import { Children } from "~/utils/typing/children";
+import { SPACE_UNIT } from "~/utils/constants";
+import { Box } from "~/index";
+
+export type ContainerProps = PropsWithChildren<{
+  "data-test-id"?: string;
+  rows: number;
+  cols: number;
+  rowsGap?: number;
+  colsGap?: number;
+  className?: string;
+  style?: CSSProperties;
+}> & {
+  children: React.ReactElement | React.ReactElement[];
```

Comment:
It looks very strange to me, you are using `PropsWithChildren` @ L18 and adding childrens here ...

Reply:
Indeed it is, the `PropsWithChildren` type implementation is: `type PropsWithChildren<P = unknown> = P & { children?: ReactNode | undefined };` so you can override the `children` type. I try to deal the type error on `child.ref?.current.dataset.check` (Line 49) but still have red underline. However, the component works correctly